extensions/git-directory-explorer/src/git-directory-explorer.tsx

```diff
@@ -0,0 +1,198 @@
+import { List, ActionPanel, Action, getPreferenceValues, showToast, Toast, Application, Icon } from "@raycast/api";
+import { useState, useEffect } from "react";
+import fs from "fs";
+import path from "path";
+import os from "os";
+import simpleGit from "simple-git";
+
+interface Preferences {
+  directory: string;
+  codeEditor: Application;
+  terminalApp: Application;
+}
+
+function expandHomeDir(directory: string): string {
+  if (directory.startsWith("~")) {
+    return path.join(os.homedir(), directory.slice(1));
+  }
+  return path.resolve(directory);
+}
+
+async function isGitRepository(dir: string): Promise<boolean> {
+  const gitDir = path.join(dir, ".git");
+  try {
+    const stats = await fs.promises.stat(gitDir);
+    return stats.isDirectory();
+  } catch {
+    return false;
+  }
+}
+
+async function getSubdirectories(rootDir: string): Promise<string[]> {
+  const entries = await fs.promises.readdir(rootDir);
+  const directories = await Promise.all(
+    entries.map(async (entry) => {
+      const fullPath = path.join(rootDir, entry);
+      const entryStat = await fs.promises.stat(fullPath);
+      return entryStat.isDirectory() ? fullPath : null;
+    }),
+  );
```

Comment:
Could maybe be one function, something like (didn't test it)

```typescript
async function getGitSubdirectories(rootDir: string): Promise<string[]> {
  const entries = await fs.promises.readdir(rootDir);
  const gitDirectories = await Promise.all(
    entries.map(async (entry) => {
      const fullPath = path.join(rootDir, entry);
      const entryStat = await fs.promises.stat(fullPath);
      if (entryStat.isDirectory()) {
        const gitDir = path.join(fullPath, ".git");
        try {
          const stats = await fs.promises.stat(gitDir);
          if (stats.isDirectory()) {
            return fullPath;
          }
        } catch {
          // Not a git directory
        }
      }
      return null;
    })
  );
  return gitDirectories.filter(Boolean) as string[];
}
```

Reply:
Estou fazendo as validações com repositórios com numero de commits maiores, em breve lanço uma atualização para correção das falhas que foram reportadas por você.